src/components/CreateScenarioButton/CreateScenarioButtonHook.js

```diff
@@ -43,6 +43,19 @@ export const useCreateScenarioButton = ({ disabled, onScenarioCreated }) => {
     return runTemplates.filter((rt) => runTemplateFilter.includes(rt.id));
   }, [solution?.data?.runTemplates, workspaceData?.solution?.runTemplateFilter]);
 
+  const filteredDatasetList = useMemo(() => {
+    const datasetFilter = workspaceData?.webApp?.options?.datasetFilter;
+    if (!datasetFilter) {
+      return datasetListData;
+    }
+
+    if (!datasetFilter.length) {
+      return [];
+    }
```

Comment:
If `datasetFilter` is invalid (not an array), we may as well return the whole `datasetListData`, don't you think ? Because an empty list of datasets will cause the webapp to be barely usable (it'll be impossible to create scenarios).
We could also print a console warning for integrators to explain that this value should be an array